lib/logstash/plugin_mixins/checked_count_logger.rb

```diff
@@ -0,0 +1,35 @@
+
+module LogStash module PluginMixins
+  class CheckedCountLogger
+    def initialize(logger)
+      @logger = logger
+      @needs_check = true
+      @count_is_supported = false
+      @in_debug = @logger.debug?
+    end
+
+    def log_statement_parameters(query, statement, parameters)
+      return unless @in_debug
+      check_count_query(query) if @needs_check
+      if @count_is_supported
+        @logger.debug("Executing JDBC query", :statement => statement, :parameters => parameters, :count => query.count)
```

Comment:
since `execute_count` was implemented to do `query.count`, maybe we use it here as well?